src/components/TextInputFocusable/index.js

```diff
@@ -220,9 +220,13 @@ class TextInputFocusable extends React.Component {
         const propStyles = StyleSheet.flatten(this.props.style);
         propStyles.outline = 'none';
         const propsWithoutStyles = _.omit(this.props, 'style');
-        const propsWithoutValueOrDefaultValue = this.props.defaultValue
-            ? _.omit(propsWithoutStyles, 'value')
-            : _.omit(propsWithoutStyles, 'defaultValue');
+        let propsWithoutValueOrDefaultValue = propsWithoutStyles;
+        if (this.props.defaultValue || !this.props.value) {
+            propsWithoutValueOrDefaultValue = _.omit(propsWithoutValueOrDefaultValue, 'value');
+        }
+        if (!this.props.defaultValue) {
+            propsWithoutValueOrDefaultValue = _.omit(propsWithoutValueOrDefaultValue, 'defaultValue');
+        }
```

Comment:
I'm not really following this...

- If we have a `defaultValue` then we remove the `value` prop
- If we have no `value` or an empty string then we remove the `value` prop
- If we have no `defaultValue` then we remove the `defaultValue`

This is confusing to me since...

1. If we have a `defaultValue` there should be no issues letting this fall through to the component. It's unclear what removing the `value` prop here achieves?
2. If `value` is an empty string then we will set it to `undefined`. Seems incorrect as it _should_ be allowed to be an empty string?
3. Again, this could be a `defaultValue` of an empty string which would then end up as `undefined`. Is that the same thing? I feel like there is a difference but can't quite remember.

What would happen if we just do nothing in this case and leave it to whoever is using the component to be responsible for `value` and `defaultValue`? Can they both just be not required props and default to `undefined`.

Reply:
Created a branch to showcase a solution I think is more correct, but lmk if there are problems with it that I'm not seeing...

https://github.com/Expensify/Expensify.cash/compare/marcaaron-fixTextInputDefaultValue